add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt

```diff
@@ -0,0 +1,76 @@
+package dev.flutter.multipleflutters
+
+import android.content.Intent
+import android.os.Bundle
+import android.widget.FrameLayout
+import android.widget.LinearLayout
+import androidx.fragment.app.FragmentActivity
+import androidx.fragment.app.FragmentManager
+import io.flutter.embedding.android.FlutterFragment
+import io.flutter.embedding.engine.FlutterEngineCache
+
+/**
+ * An activity that displays 2 FlutterFragments vertically.
+ */
+class DoubleFlutterActivity : FragmentActivity(), EngineBindingsDelegate {
+    private val topBindings : EngineBindings by lazy {
+        EngineBindings(this, this, "topMain")
+    }
+    private val bottomBindings : EngineBindings by lazy {
+        EngineBindings(this, this, "bottomMain")
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val numberOfFlutters = 2
+        val root = LinearLayout(this)
+        root.layoutParams = LinearLayout.LayoutParams(
+            LinearLayout.LayoutParams.MATCH_PARENT,
+            LinearLayout.LayoutParams.MATCH_PARENT
+        )
+        root.orientation = LinearLayout.VERTICAL
+        root.weightSum = numberOfFlutters.toFloat()
+
+        val fragmentManager: FragmentManager = supportFragmentManager
+
+        setContentView(root)
+
+        val app : App = applicationContext as App
+
+        for (i in 0 until numberOfFlutters) {
+            val flutterContainer = FrameLayout(this)
+            root.addView(flutterContainer)
+            flutterContainer.id = 12345 + i
+            flutterContainer.layoutParams = LinearLayout.LayoutParams(
+                FrameLayout.LayoutParams.MATCH_PARENT,
+                FrameLayout.LayoutParams.MATCH_PARENT,
+                1.0f
+            )
+            val engine = if (i == 0) topBindings.engine else bottomBindings.engine
+            FlutterEngineCache.getInstance().put(i.toString(), engine)
+            val flutterFragment =
+                FlutterFragment.withCachedEngine(i.toString()).build<FlutterFragment>()
+            fragmentManager
+                .beginTransaction()
+                .add(
+                    12345 + i,
+                    flutterFragment
+                )
+                .commit()
+        }
+
+        topBindings.attach()
```

Comment:
Could we do this for iOS too for consistency?

Reply:
It already is happening, but in a slightly different way.  On iOS SingleFlutterViewController bundles the bindings with itself so it's not necessary.  I wouldn't want to add a concept that isn't necessary.  I don't think that would make the sample more clear.

A more similar approach would be to have Fragment that bundles the communication with a FlutterFragment, then have SigqleFlutterActivity and DoubleFlutterActivity embed that Fragment.  Unfortunately for me, converting an Activity to a Fragment isn't trivial.

Reply:
Ya, I just meant generally where the platform channel handlers sit between the 2 platforms if someone was looking for them in each of the platforms. If they need to be in different sounding files, maybe add a comment cross-referencing the difference?